src/vaev-engine/layout/flex.cpp

```diff
@@ -1423,17 +1438,23 @@ struct FlexFormatingContext : FormatingContext {
             for (auto& flexItem : flexLine.items) {
                 flexItem.position = flexItem.position + flexLine.position + input.position;
 
-                auto out = layout(
-                    tree,
-                    *flexItem.box,
-                    {
-                        .fragment = input.fragment,
-                        .knownSize = {flexItem.usedSize.x, flexItem.usedSize.y},
-                        .position = flexItem.position,
-                        .availableSpace = availableSpace,
-                    }
-                );
-                flexItem.commit(input.fragment);
+                UsedSpacings usedSpacings{
+                    .padding = std::move(flexItem.padding),
+                    .borders = std::move(flexItem.borders),
+                    .margin = flexItem.resolvedMargin(),
+                };
+
+                Input childInput{
+                    .fragment = input.fragment,
+                    .knownSize = {flexItem.usedSize.x, flexItem.usedSize.y},
+                    .position = flexItem.position,
+                    .availableSpace = availableSpace,
+                };
+
+                if (input.fragment)
+                    layoutAndCommitBorderBox(tree, *flexItem.box, childInput, *input.fragment, usedSpacings);
+                else
+                    layoutBorderBox(tree, *flexItem.box, childInput, usedSpacings);
```

Comment:
The parent function should only be called during commit, so there will always be a fragment.

Reply:
t'as raison! merci